lnst/Tests/Iperf.py

```diff
@@ -32,59 +32,47 @@ def run(self):
         server = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE,
                                   stderr=subprocess.PIPE, close_fds=True)
 
-        try:
-            stdout, stderr = server.communicate()
-            stdout = stdout.decode()
-            stderr = stderr.decode()
-        except KeyboardInterrupt:
```

Comment:
i think this `KeyboardInterrupt` should be kept for at least the server side (though i don't see why it couldn't be the same for client side as well) of iperf as there may be situations where there may be a usecase to do `server_job.kill(signal.SIGINT)` to gracefully exit the process.

yes, we don't do this in the `IperfFlowMeasurement`, but this is a Test module which may be used in different usecases.

Reply:
I brought it back with `server.wait()` and put `.communicate()` into the `finally` block. There might be some potential buffering issues, but I don't understand it fully yet. I'll have to test this I guess. The alternative to this would be to copy the `communicate()` line into the `try` block as well (I think).

Reply:
> `Popen.wait()` Note: This will deadlock when using stdout=PIPE or stderr=PIPE and the child process generates enough output to a pipe such that it blocks waiting for the OS pipe buffer to accept more data. Use [Popen.communicate()](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate) when using pipes to avoid that.

We should be using `.communicate()` definitely. I changed it so. Should be good now.